### ISSUE TITLE
useWindowSize does not work on App Router (nextjs)

### ISSUE THREAD

**Issue by ghostlexly** [streamich/react-use#2532]
**What is the current behavior?**

Gives an error: 
app-index.js:32 Warning: Prop `width` did not match. Server: "Infinity" Client: "1290"

This is probably due to app router running the code on server side even with "use client" specified.

**Steps to reproduce it and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have extra dependencies other than `react-use`. Paste the link to your [JSFiddle](https://jsfiddle.net) or [CodeSandbox](https://codesandbox.io) example below:**

- install nextjs 14
- use app router
- create a component

`tsx
"use client";

import useWindowSize from "react-use/lib/useWindowSize";

const StepSubmittedWidget = () => {
  const { width, height } = useWindowSize();

return (<></>)
}

export default StepSubmittedWidget;
`

**What is the expected behavior?**

Gives the browser's window sizes

**A little about versions:**  
- _OS_:  MacOS
- _Browser (vendor and version)_:  Chrome
- _React_: 18.2.0
- _`react-use`_: 17.4.2
- _Did this worked in the previous package version?_ No

**Comment by vincentjr** [comment]
Running into the same issue here :x

**Comment by geanify** [comment]
I don't think this is an issue, if you are doing Server Side Rendering for that component, it won't have a screen size to fetch. 

Don't use a server component, try to use a client component to fix this issue

See: 
https://nextjs.org/docs/app/building-your-application/rendering/server-components
https://nextjs.org/docs/app/building-your-application/rendering/client-components

**Comment by childrentime** [comment]
try https://reactuse.com/element/useWindowSize/
